OpenRA.Mods.Common/Traits/Buildings/RallyPoint.cs

```diff
@@ -99,6 +113,17 @@ public bool CanTarget(Actor self, Target target, List<Actor> othersAtTarget, ref
 				if (self.World.Map.Contains(location))
 				{
 					cursor = "ability";
+
+					// Set also PrimaryBuilding with Ctrl and only this one building of the same type selected
+					if (modifiers.HasModifier(TargetModifiers.ForceAttack))
+					{
+						var selfName = self.Info.Name;
+						if (self.World.Selection.Actors.Count(a => a.Info.Name == selfName) == 1)
+						{
+							SetPrimaryBuilding = true;
+						}
+					}
+
```

Comment:
Since you never set SetPrimaryBuilding to false anywhere wouldn't this mean that if you control click once with a selected building every subsequent click while you don't replace the OrderTargeter will set the primary building.
(Might not matter since it does not change anything when you set a primary building to be a primary building).